src/dotty/tools/dotc/transform/LambdaLift.scala

```diff
@@ -195,17 +245,16 @@ class LambdaLift extends MiniPhase with IdentityDenotTransformer { thisTransform
         }
         tree match {
           case tree: Ident =>
-            if (sym.maybeOwner.isTerm) {
+            if (isLocal(sym)) {
               if (sym is Label)
                 assert(enclosure == sym.enclosure,
                   i"attempt to refer to label $sym from nested $enclosure")
               else if (sym is Method) markCalled(sym, enclosure)
               else if (sym.isTerm) markFree(sym, enclosure)
-            } else if (sym.maybeOwner.isClass)
-              narrowTo(sym.owner.asClass)
+            }
+            if (sym.maybeOwner.isClass) narrowTo(sym.owner.asClass)
```

Comment:
I'm asking because it would be kind of nice to add `if sym.isTerm && isLocal(sym)` as a guard to the Ident case (and `sym.isMethod && isLocal(sym)`) and perhaps treat the narrowing separately.